modules/svg/src/SkSVGFilter.cpp

```diff
@@ -42,12 +46,12 @@ sk_sp<SkImageFilter> SkSVGFilter::buildFilterDAG(const SkSVGRenderContext& ctx)
         // color-interpolation-filters). We call this explicitly here because the SkSVGFe
         // nodes do not participate in the normal onRender path, which is when property
         // propagation currently occurs.
-        SkSVGRenderContext localCtx(ctx);
-        feNode.applyProperties(&localCtx);
+        SkSVGRenderContext localChildCtx(localCtx);
+        feNode.applyProperties(&localChildCtx);
 
-        const SkRect filterSubregion = feNode.resolveFilterSubregion(localCtx, fctx);
-        cs = feNode.resolveColorspace(ctx, fctx);
```

Comment:
The colorspace was resolved from the original context (`ctx`) rather than the local one. This solves the problem shown in Example 2 in the description.